### PR TITLE
Fix error message bubbling up on seed error

### PR DESCRIPTION
**Pull request by risseraka**
Problem: When the nth seed `throw`s an error, all subsequent seeds add an error message on top of original message which makes it less clear which seed the error originated from.

Solution: Nest promises and `catch` error inside the nested seed promise.

Before:
```
Error: Error while executing ".../knex/seeds/9-titres-activites.js" seed: Error while executing ".../knex/seeds/8-metas-activites.js" seed: Error while e
xecuting ".../knex/seeds/7-titres.js" seed: insert into "titres_points" ("contour", "coordonnees", "description", "groupe", "id", "nom", "point"
    at Object.current.then.then.catch.originalError (.../node_modules/knex/lib/seed/Seeder.js:166:23)
Error: Error while executing ".../knex/seeds/8-metas-activites.js" seed: Error while executing ".../knex/seeds/7-titres.js" seed: insert into "titres_poi
nts" ("contour", "coordonnees", "description", "groupe", "id", "nom", "point"
    at Object.current.then.then.catch.originalError (.../node_modules/knex/lib/seed/Seeder.js:166:23)
Error: Error while executing ".../knex/seeds/7-titres.js" seed: insert into "titres_points" ("contour", "coordonnees", "description", "groupe", "id", "nom", "point" [...]
    at Object.current.then.then.catch.originalError (.../node_modules/knex/lib/seed/Seeder.js:166:23)
```

After:
```
Error: Error while executing ".../knex/seeds/7-titres.js" seed: insert into "titres_points" ("contour", "coordonnees", "description", "groupe", "id", "nom", "point" [...]
    at Object.current.then.then.catch.originalError (.../node_modules/knex/lib/seed/Seeder.js:166:23)
```

LMKWYT,
Cheers.

### CODE REVIEW

**Comment by risseraka** [comment]
Hi Igor,

Sure, this behavior is not trivial, but I'll try to come up with something.

**Comment by risseraka** [comment]
Alright, I've added a straightforward test, which fails on master but passes on this branch.

Error output on master:
```
Error while executing "[...]/knex/test/unit/seed/test/2-second.js" seed: Error while executing "[...]/knex/test/unit/seed/test/1-first.js" seed: throwing in first file
```

Error output on this branch:
```
Error while executing "[...]/knex/test/unit/seed/test/1-first.js" seed: throwing in first file
```

PS: tests seem to be quite broken on the past few pull requests, is it so;ething to be worried about?

File: test/unit/seed/seeder.js [hunk 1]
```diff
@@ -67,3 +67,33 @@ describe('Seeder.loadExtensions', function() {
...
+
+  beforeEach(function() {
+    seeder = knex(config).seed;
+  });
+
+  it('should throw an error with correct file name', () => {
+    return seeder
+      ._waterfallBatch(['1-first.js', '2-second.js'])
+      .catch((error) => {
+        expect(error.message).to.match(
```
Comment (kibertoad) [inline test/unit/seed/seeder.js]: This could use done() callback call, otherwise test will pass even if no errors are thrown.

Comment (risseraka) [reply test/unit/seed/seeder.js]: Good catch, fixing right away.
(I've added missing test files too.)

Comment (risseraka) [reply test/unit/seed/seeder.js]: Done, resolving conversation.

**Comment by kibertoad** [comment]
@risseraka There are some flaky Oracle tests, these are to be ignored, anything else - I'd need to check that out :)

**Comment by kibertoad** [comment]
Thanks a lot for your contribution!

**Comment by risseraka** [comment]
You're very welcome, thanks for your and every contributor's hard work.

(Impressive speed by the way. 😅)

**Comment by kibertoad** [comment]
My pleasure!
How urgently do you need this change released? I wasn't planning another release before 0.18 unless something serious popped up, but we can figure something out if it's something you need urgently.

**Comment by risseraka** [comment]
@kibertoad No worries, there's absolutely no urgency, it was just a bit annoying to look at logs on errors, so it's mostly cosmetic, take your time to release whenever you see fit.

**Comment by kibertoad** [comment]
Released in 0.18.0-next1

**Comment by risseraka** [comment]
Thanks for the heads up.
Hurray for @kibertoad!

**Pull request merged**